ccd-adapter/src/main/java/uk/gov/hmcts/cmc/ccd/domain/mapper/claimantresponse/ClaimantResponseMapper.java

```diff
@@ -0,0 +1,90 @@
+import uk.gov.hmcts.cmc.domain.models.claimantresponse.ResponseRejection;
+import uk.gov.hmcts.cmc.domain.models.response.YesNoOption;
+
+@Component
+public class ClaimantResponseMapper {
+
+    private final PaymentIntentionMapper paymentIntentionMapper;
+
+    @Autowired
+    public ClaimantResponseMapper(PaymentIntentionMapper paymentIntentionMapper) {
+        this.paymentIntentionMapper = paymentIntentionMapper;
+    }
+
+    public CCDClaimantResponse to(Claim claim) {
+        final ClaimantResponse claimantResponse = claim.getClaimantResponse().orElse(null);
+        if (null == claimantResponse) {
+            return null;
+        }
+
+        if (ClaimantResponseType.ACCEPTATION == claimantResponse.getType()) {
+            ResponseAcceptation responseAcceptation = (ResponseAcceptation) claimantResponse;
+            CCDResponseAcceptation.CCDResponseAcceptationBuilder builder = CCDResponseAcceptation.builder();
+            responseAcceptation.getAmountPaid().ifPresent(builder::amountPaid);
+            responseAcceptation.getFormaliseOption()
+                .map(FormaliseOption::name)
+                .map(CCDFormaliseOption::valueOf)
+                .ifPresent(builder::formaliseOption);
+            responseAcceptation.getClaimantPaymentIntention().ifPresent(
+                paymentIntention -> builder.claimantPaymentIntention(paymentIntentionMapper.to(paymentIntention))
+            );
+            return builder.build();
+        } else if (ClaimantResponseType.REJECTION == claimantResponse.getType()) {
+            ResponseRejection responseRejection = (ResponseRejection) claimantResponse;
+            CCDResponseRejection.CCDResponseRejectionBuilder rejection = CCDResponseRejection.builder()
+                .freeMediationOption(CCDYesNoOption.valueOf(YesNoOption.NO.name()));
+            responseRejection.getAmountPaid().ifPresent(rejection::amountPaid);
+            responseRejection.getReason().ifPresent(rejection::reason);
+            return rejection.build();
+        }
+        throw new MappingException("unsupported claimant response type " + claimantResponse.getType());
+    }
+
+    public Claim from(CCDClaimantResponse ccdClaimantResponse) {
+        if (null == ccdClaimantResponse) {
+            return null;
+        }
+        Claim.ClaimBuilder claimBuilder = Claim.builder();
+        if (ccdClaimantResponse.getClaimantResponseType() == CCDClaimantResponseType.ACCEPTATION) {
+            CCDResponseAcceptation ccdResponseAcceptation = (CCDResponseAcceptation) ccdClaimantResponse;
+            claimBuilder.claimantResponse(ResponseAcceptation.builder()
+                .amountPaid(ccdResponseAcceptation.getAmountPaid())
+                .formaliseOption(FormaliseOption.valueOf(ccdResponseAcceptation.getFormaliseOption().name()))
+                .claimantPaymentIntention(paymentIntentionMapper.from(ccdResponseAcceptation
+                    .getClaimantPaymentIntention()))
```

Comment:
above to method got Optional check for claimant Payment Intention but from doesn't have any. Neither
PaymentIntentionMapper has got null check, will throw null pointer.

Reply:
have added the check in PaymentIntentionMapper